p2p/src/test/kotlin/maru/p2p/BeaconBlocksByRangeIntegrationTest.kt

```diff
@@ -0,0 +1,285 @@
+    val maruPeer1 =
+      DefaultMaruPeer(
+        peer1,
+        createRpcMethods(beaconChain2),
+        StatusMessageFactory(beaconChain2, createForkIdHashProvider(beaconChain2)),
+      )
+
+    // Send request for blocks
+    val request = BeaconBlocksByRangeRequest(startBlockNumber = 3UL, count = 5UL)
+    val requestMessage =
+      Message(
+        type = RpcMessageType.BEACON_BLOCKS_BY_RANGE,
+        version = Version.V1,
+        payload = request,
+      )
+
+    val responseHandler = MaruRpcResponseHandler<Message<BeaconBlocksByRangeResponse, RpcMessageType>>()
+
+    peer1.sendRequest(
+      createRpcMethods(beaconChain2).beaconBlocksByRange(),
+      requestMessage,
+      responseHandler,
+    )
+
+    val responseMessage = responseHandler.response().get(5, TimeUnit.SECONDS)
+    val response = responseMessage.payload
+
+    // Verify response
+    assertThat(response.blocks).hasSize(5)
+    assertThat(
+      response.blocks[0]
+        .beaconBlock.beaconBlockHeader.number,
+    ).isEqualTo(3UL)
+    assertThat(
+      response.blocks[1]
+        .beaconBlock.beaconBlockHeader.number,
+    ).isEqualTo(4UL)
+    assertThat(
+      response.blocks[2]
+        .beaconBlock.beaconBlockHeader.number,
+    ).isEqualTo(5UL)
+    assertThat(
+      response.blocks[3]
+        .beaconBlock.beaconBlockHeader.number,
+    ).isEqualTo(6UL)
+    assertThat(
+      response.blocks[4]
+        .beaconBlock.beaconBlockHeader.number,
+    ).isEqualTo(7UL)
+
+    // Verify the actual block content matches
+    assertThat(response.blocks[0]).isEqualTo(storedBlocks[2]) // block 3 is at index 2
+    assertThat(response.blocks[1]).isEqualTo(storedBlocks[3])
+    assertThat(response.blocks[2]).isEqualTo(storedBlocks[4])
+    assertThat(response.blocks[3]).isEqualTo(storedBlocks[5])
+    assertThat(response.blocks[4]).isEqualTo(storedBlocks[6])
+  }
+
+  @Test
+  fun `peer receives empty response when requesting non-existent blocks`() {
+    // Start networks
+    p2pNetwork1.start().get()
+    p2pNetwork2.start().get()
+
+    // Wait for peers to connect
+    awaitUntilAsserted {
+      assertThat(p2pNetwork1.peerCount).isEqualTo(1)
+      assertThat(p2pNetwork2.peerCount).isEqualTo(1)
+    }
+
+    // Get peer reference
+    val peer1 =
+      p2pNetwork2.peerLookup.getPeer(LibP2PNodeId(PeerId.fromBase58(PEER_ID_NODE_1)))
+        ?: throw IllegalStateException("Peer not found")
+
+    // Send request for blocks beyond what exists
+    val request = BeaconBlocksByRangeRequest(startBlockNumber = 100UL, count = 10UL)
+    val requestMessage =
+      Message(
+        type = RpcMessageType.BEACON_BLOCKS_BY_RANGE,
+        version = Version.V1,
+        payload = request,
+      )
+
+    val responseHandler = MaruRpcResponseHandler<Message<BeaconBlocksByRangeResponse, RpcMessageType>>()
+
+    peer1.sendRequest(
+      createRpcMethods(beaconChain2).beaconBlocksByRange(),
+      requestMessage,
+      responseHandler,
+    )
```

Comment:
I think this part is too low level to be tested here and to be exposed to the user's code. Maybe we should instead instantiate all the necessary stuff with the same factory that will be used by Maru itself?

And will request be sent with `peer1.sendRequest` or there will be more high-level interface to interact with?